widgets/src/command_text_input.rs

```diff
@@ -267,28 +255,36 @@ impl CommandTextInput {
         cx.widget_action(self.widget_uid(), &scope.path, InternalAction::ItemSelected);
         self.hide_popup();
         self.is_text_input_focus_pending = true;
-        self.try_remove_trigger_char();
+        self.try_remove_trigger_grapheme();
         self.redraw(cx);
     }
 
-    fn try_remove_trigger_char(&mut self) {
-        let text_input = self.text_input_ref();
-        let cursor_pos = text_input.borrow().map_or(0, |p| p.get_cursor().head.index);
-        if cursor_pos > 0 {
-            let last_char_pos = cursor_pos - 1;
-            let last_char = text_input.text().chars().nth(last_char_pos);
+    fn try_remove_trigger_grapheme(&mut self) {
+        let head = get_head(&self.text_input_ref());
 
-            if last_char == self.trigger_char() {
-                let at_removed = text_input
-                    .text()
-                    .chars()
-                    .enumerate()
-                    .filter_map(|(i, c)| if i == last_char_pos { None } else { Some(c) })
-                    .collect::<String>();
+        if head == 0 {
+            return;
+        }
 
-                text_input.set_text(&at_removed);
-                self.previous = at_removed;
-            }
+        let text = self.text();
+        let Some((inserted_grapheme_pos, inserted_grapheme)) =
+            inserted_grapheme_with_pos(&text, head)
+        else {
+            return;
+        };
+
+        if Some(inserted_grapheme) == self.trigger_grapheme() {
```

Comment:
nit: it's more readable and Rust-y to flip it, otherwise it looks like a mistaken `if let Some(...)`.
```suggestion
        if self.trigger_grapheme() == Some(inserted_grapheme) {
```